core/src/main/java/hudson/views/GlobalDefaultViewConfiguration.java

```diff
@@ -51,7 +51,9 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
             j.setPrimaryView(newPrimaryView);
         } else {
             // Fallback if the view is not specified
-            j.setPrimaryView(j.getViews().iterator().next());
+            if(j.getViews().iterator().hasNext()) {
```

Comment:
I had an error page using a user with CONFIGURE permission because this was null and fires a NullPointerException. I don't remember which page exactly, I can test again. But maybe the issue is that the user with CONFIGURE should have a primaryView set instead ? I add this to the todo list.

Reply:
Related to #4348?

Reply:
You should store the Iterator in a variable. Technically, checking hasNext() and next() on two separate iterators doesn't make sense.